### ISSUE TITLE
Add Data Wrangler to flat Python file

### ISSUE THREAD

**Issue by BrazilForever11** [microsoft/vscode-data-wrangler#299]
This is not a bug. This is feature request. 

It would be great to add Data Wrangler to flat Python file (not just Jupyter notebook). 

Here is justification for working with flat files vs notebooks: 
- I often need to execute just a single line of code. I do not need to run whole cell. Creating a cell for each separate line eats up screen space.
-  Individual cell's output again eats up a lot of screen space. I would rather have all my output in the terminal.

P.S.
A bit off topic, but in the long term it would be great to be able to create markdown cell on demand, rather than having all code in Jupyter cells.

**Comment by pwang347** [comment]
Hi @BrazilForever11, thanks for opening this issue! We do support launching from the Python debugger for .py files, you can access this through the following menu:
<img width="386" alt="image" src="https://github.com/user-attachments/assets/a8894c29-b7cd-4958-91a2-b75181d51ac9">

If you want to do this line-by-line, have you tried using the following command (first run `ctrl+shift+P` to open command palette), and then type the following:
<img width="728" alt="image" src="https://github.com/user-attachments/assets/82f48a98-70f0-4566-bf31-673dd215ff54">

Does this work for you, or were you perhaps looking for something else (e.g., some special Python code to launch a new window when executed, but not debugging)?

**Comment by BrazilForever11** [comment]
Thank you so much! 

Using debugger is an interesting idea. I will think more about it. 

But originally I was thinking of something similar to R-studio, where I can run code (flat file) line by line and as I run lines variables show up in Variable Explorer where I can view them. This is how R-Studio normally operate (it does not involve debugger)

**Comment by pwang347** [comment]
No problem! I think line-by-line execution in a .py file could be an interesting idea. Though if implemented as you described, it would likely need to be done from the Jupyter extension since they own the variable panel UI (which Data Wrangler taps into via a public data viewer API). They also have a similar command in Jupyter notebooks, which I'm guessing depends on IPython:

<img width="623" alt="image" src="https://github.com/user-attachments/assets/55fa20c2-cbaa-4603-9997-1038c14bf9bb">

**Comment by BrazilForever11** [comment]
I think it is already possible to execute .py file line-by-line in VS Code.  Shift + Enter executes a line and results stay in memory. I think it is only matter of reading it from memory and displaying it with Data Wrangler (variable explorer)

**Comment by pwang347** [comment]
Makes sense. Seems like what you are looking for is actually something like this (variable explorer in general, with integration to DW): https://github.com/microsoft/vscode-python/issues/17183

I see that it is being considered as part of a recent plan here, so this may be an issue to subscribe to for any developments: https://github.com/microsoft/vscode-python/issues/21707

Will keep this issue open here as well since we will likely need to integrate with the new proposed variables panel once/if it releases. Thanks!
